### PR TITLE
#125 Report Path sanitization

### PR DESCRIPTION
**Pull request by sarmadali20**
fix the report paths when `:` is part of device id connected to adb replace with `-`

### CODE REVIEW

**Comment by sarmadali20** [comment]
https://github.com/gojuno/composer/issues/125

**Comment by sarmadali20** [comment]
quick fix may need cleanup

**Comment by sarmadali20** [comment]
@artem-zinnatullin have a look when u get a chance

File: composer/src/main/kotlin/com/gojuno/composer/Main.kt [hunk 1]
```diff
@@ -90,7 +90,7 @@ fun main(rawArgs: Array<String>) {
                                         .flatMap { adbDeviceTestRun ->
                                             writeJunit4Report(
                                                     suite = adbDeviceTestRun.toSuite(args.testPackage),
-                                                    outputFile = File(File(args.outputDirectory, "junit4-reports"), "${device.id}.xml")
+                                                    outputFile = File(File(args.outputDirectory, "junit4-reports"), "${device.sanitizedId()}.xml")
```
Comment (artem-zinnatullin) [inline composer/src/main/kotlin/com/gojuno/composer/Main.kt]: hm, this might break someones report parsing, we should emphasize this in release notes

File: composer/src/main/kotlin/com/gojuno/composer/Main.kt [hunk 2]
```diff
@@ -190,7 +190,9 @@ data class Device(
         val id: String,
         val logcat: File,
         val instrumentationOutput: File
-)
+){
+    fun sanitizedId() = id.replace(":","-")
```
Comment (artem-zinnatullin) [inline composer/src/main/kotlin/com/gojuno/composer/Main.kt]: Probably a property would look better, but I don't have strong opinion :)

File: composer/src/main/kotlin/com/gojuno/composer/TestRun.kt [hunk 3]
```diff
@@ -38,6 +42,8 @@ data class AdbDeviceTest(
     }
 }
 
+fun AdbDevice.sanitizedId() = id.replace(":","-")
```
Comment (artem-zinnatullin) [inline composer/src/main/kotlin/com/gojuno/composer/TestRun.kt]: That's not good, we shouldn't duplicate implementation, it'll be easy to de-sync them

File: composer/src/main/kotlin/com/gojuno/composer/Main.kt (hunk 2 above)
Comment (artem-zinnatullin) [reply composer/src/main/kotlin/com/gojuno/composer/Main.kt]: We need tests btw :)

File: composer/src/main/kotlin/com/gojuno/composer/Main.kt (hunk 1 above)
Comment (koral--) [reply composer/src/main/kotlin/com/gojuno/composer/Main.kt]: There is also an edge case when filenames for multiple devices will be equal after sanitization e.g. both `123:456` and `123-456`.

**Comment by sarmadali20** [comment]
I will work on the feedback once I find some time, just busy with building new infra so might not need this anymore cuz my Jenkins master is now a Linux machine :)
